Documentation/doc/CMakeLists.txt

```diff
@@ -0,0 +1,319 @@
+    endif()
+    if(IS_DIRECTORY ${CMAKE_SOURCE_DIR}/${pkg}/include)
+      set(CGAL_${pkg}_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/${pkg}/include")
+    endif()
+    if(IS_DIRECTORY ${CMAKE_SOURCE_DIR}/${pkg}/include)
+      set(CGAL_${pkg}_DEMO_DIR "${CMAKE_SOURCE_DIR}/${pkg}/demo")
+    endif()
+  endforeach()
+else()
+  subdirlist(CGAL_DOC_PACKAGES ${CMAKE_SOURCE_DIR}/doc)
+  # Remove known offenders
+  list(REMOVE_ITEM CGAL_DOC_PACKAGES "scripts" "biblio" "resources" "Documentation")
+  subdirlist(CGAL_EXAMPLE_PACKAGES ${CMAKE_SOURCE_DIR}/examples)
+  subdirlist(CGAL_DEMO_PACKAGES ${CMAKE_SOURCE_DIR}/demo)
+  foreach(pkg ${CGAL_EXAMPLE_PACKAGES})
+    set(CGAL_${pkg}_EXAMPLE_DIR "${CMAKE_SOURCE_DIR}/examples")
+    # This might not be entirely correct, but there is no reliable,
+    # easy way to get a list of all package names when not in a
+    # branch build.
+    set(CGAL_${pkg}_INCLUDE_DIR "${CMAKE_SOURCE_DIR}/include")
+  endforeach()
+  foreach(pkg ${CGAL_DEMO_PACKAGES})
+    set(CGAL_${pkg}_DEMO_DIR "${CMAKE_SOURCE_DIR}/demo")
+  endforeach()
+endif()
+
+foreach(package ${CGAL_DOC_PACKAGES})
+  configure_doxygen_package(${package})
+endforeach()
+
+# Add a custom target "doc"
+add_custom_target(doc)
+add_custom_target(doc_pre)
+add_custom_target(doc_post)
+
+# do the main package manually, it isn't part of ${CGAL_CONFIGURED_PACKAGES_NAMES}
+configure_doxygen_package("Documentation")
+set_target_properties(Documentation_doc PROPERTIES FOLDER Documentation)
+
+# hard-code that doc depends on all packages
+foreach(depend ${CGAL_DOC_PACKAGES})
+  add_dependencies(doc_pre ${depend}_internal_doxygen_run)
+  add_dependencies(doc_post ${depend}_copy_doc_tags)
+  add_dependencies(${depend}_copy_doc_tags doc_pre)
+endforeach()
+add_dependencies(doc_pre Documentation_internal_doxygen_run)
+add_dependencies(doc_post Documentation_copy_doc_tags)
+add_dependencies(Documentation_copy_doc_tags doc_pre)
+
+#total level doc dependencies
+add_dependencies(doc doc_post)
+
+if(PYTHONINTERP_FOUND)
+  set(CGAL_DOC_TESTSUITE_SCRIPT "${CGAL_DOC_SCRIPT_DIR}/testsuite.py")
+
+  add_custom_target(doc_with_postprocessing
+    ${PYTHON_EXECUTABLE} ${CGAL_DOC_SCRIPT_DIR}/html_output_post_processing.py --output ${CGAL_DOC_OUTPUT_DIR} --resources ${CGAL_DOC_RESOURCE_DIR}
+  )
+  add_dependencies(doc_with_postprocessing doc)
+
+  if(CGAL_DOC_CREATE_LOGS)
+    add_custom_target(Documentation_test
+      ${PYTHON_EXECUTABLE} ${CGAL_DOC_TESTSUITE_SCRIPT} --output-dir ${CGAL_DOC_OUTPUT_DIR} --doc-log-dir ${CGAL_DOC_LOG_DIR}
+    )
+    add_dependencies(Documentation_test doc)
+
+    add_custom_target(Documentation_test_publish
+      ${PYTHON_EXECUTABLE} ${CGAL_DOC_TESTSUITE_SCRIPT} --output-dir ${CGAL_DOC_OUTPUT_DIR} --doc-log-dir ${CGAL_DOC_LOG_DIR} --publish ${CGAL_DOC_PUBLISH_DIR} --do-copy-results
+    )
+    add_dependencies(Documentation_test_publish doc)
+
+    set(CGAL_DOC_VERSION_H "/tmp/version.h" CACHE PATH "Path to CGAL/version.h")
+
```

Comment:
The version should be passed as a CMake variable, instead of using a file in `/tmp`.